src/dfx/src/lib/operations/canister/install_canister.rs

```diff
@@ -159,6 +164,52 @@ pub async fn install_canister(
         post_install_store_assets(canister_info, agent, timeout).await?;
     }
 
+    if !canister_info.get_post_install().is_empty() {
```

Comment:
A suggestion: move this all into a separate method, in part so that it can have its own `#[context]`, and in part because the rest of the parts of this method would be better refactored into smaller methods.